packages/block-editor/src/components/block-draggable/index.native.js

```diff
@@ -261,53 +248,76 @@ const BlockDraggable = ( { clientId, children } ) => {
 			BLOCK_OPACITY_ANIMATION_DELAY,
 			withTiming( 1, BLOCK_OPACITY_ANIMATION_CONFIG )
 		);
-		runOnJS( selectBlock )( clientId );
 	};
 
-	const { isDraggable, isBeingDragged } = useSelect(
+	const { isDraggable, isBeingDragged, canDragBlock } = useSelect(
 		( select ) => {
 			const {
 				getBlockRootClientId,
 				getTemplateLock,
 				isBlockBeingDragged,
+				hasSelectedBlock,
 			} = select( blockEditorStore );
 			const rootClientId = getBlockRootClientId( clientId );
 			const templateLock = rootClientId
 				? getTemplateLock( rootClientId )
 				: null;
+			const isAnyTextInputFocused =
+				TextInputState.currentlyFocusedInput() !== null;
 
 			return {
 				isBeingDragged: isBlockBeingDragged( clientId ),
 				isDraggable: 'all' !== templateLock,
+				canDragBlock: hasSelectedBlock()
+					? ! isAnyTextInputFocused
+					: true,
 			};
 		},
 		[ clientId ]
 	);
 
 	useEffect( () => {
-		if ( isBeingDragged ) {
-			startDraggingBlock();
-			wasBeingDragged.current = true;
-		} else if ( wasBeingDragged.current ) {
-			stopDraggingBlock();
-			wasBeingDragged.current = false;
+		if ( isBeingDragged !== wasBeingDragged.current ) {
+			if ( isBeingDragged ) {
+				startDraggingBlock();
+			} else {
+				stopDraggingBlock();
+			}
 		}
+		wasBeingDragged.current = isBeingDragged;
 	}, [ isBeingDragged ] );
 
-	const wrapperStyles = useAnimatedStyle( () => {
+	const animatedWrapperStyles = useAnimatedStyle( () => {
 		return {
 			opacity: draggingAnimation.opacity.value,
 		};
 	} );
+	const wrapperStyles = [
+		animatedWrapperStyles,
+		styles[ 'draggable-wrapper__container' ],
+	];
 
 	if ( ! isDraggable ) {
 		return children( { isDraggable: false } );
 	}
 
 	return (
-		<Animated.View style={ wrapperStyles }>
-			{ children( { isDraggable: true } ) }
-		</Animated.View>
+		<DraggableTrigger
+			id={ clientId }
+			enabled={ enabled && canDragBlock }
+			minDuration={ Platform.select( {
```

Comment:
What do you think if we move this to a variable? 😁

Reply:
Also can we have a const for the `450` value?

Reply:
For the `450` value, I'm considering calculating it by subtracting `50` ms from `DEFAULT_LONG_PRESS_MIN_DURATION`, as we only need to have a lower value.

Reply:
That sounds good, I think having that calculation in a variable would be good, like:

```javascript
const DEFAULT_IOS_LONG_PRESS_MIN_DURATION = DEFAULT_LONG_PRESS_MIN_DURATION - 50;
```

Or something like that 😄